engine/src/gfx/cockpit.cpp

```diff
@@ -102,749 +106,80 @@ using VSFileSystem::SoundFile;
 static soundContainer disableautosound;
 static soundContainer enableautosound;
 
-#define sqr( x ) ( ( (x) )*( (x) ) )
-std::string GameCockpit::GetNavSelectedSystem()
-{
-    return AccessNavSystem()->getSelectedSystem();
-}
+vector< int >respawnunit;
+vector< int >switchunit;
+vector< int >turretcontrol;
 
-void soundContainer::loadsound( string soundfile, bool looping )
+static float getInitialZoomFactor()
 {
-    if ( this->sound == -2 && soundfile.size() ) {
-        string sound = GameCockpit::getsoundfile( soundfile );
-        if ( sound.size() )
-            this->sound = AUDCreateSoundWAV( sound, looping );
-        else
-            this->sound = -1;
-    }
+    static float inizoom = XMLSupport::parse_float( vs_config->getVariable( "graphics", "inital_zoom_factor", "2.25" ) );
+    return inizoom;
 }
 
-void soundContainer::playsound()
+void visitSystemHelp( Cockpit *cp, string systemname, float num )
 {
-    if (sound >= 0) {
-        AUDAdjustSound( sound, QVector( 0, 0, 0 ), Vector( 0, 0, 0 ) );
-        AUDPlay( sound, QVector( 0, 0, 0 ), Vector( 0, 0, 0 ), 1 );
-    }
+    string key( string( "visited_" )+systemname );
+    vector< float > *v = &cp->savegame->getMissionData( key );
+    if ( v->empty() )
+        v->push_back( num );
+    else if ( (*v)[0] != 1.0 && num == 1 )
+        (*v)[0] = num;
 }
 
-soundContainer::~soundContainer()
+#define sqr( x ) ( ( (x) )*( (x) ) )
```

Comment:
Probably should use `pow(x, 2)` (https://www.cplusplus.com/reference/cmath/pow/) instead